docs/frontend-system/architecture/04-plugins.md

```diff
@@ -10,84 +10,151 @@ description: Frontend plugins
 
 ## Creating a Plugin
 
-<!--
+A plugin is can easily be created, it only requires a plugin `id`:
 
-How to create a simple plugin
-
- -->
-
-```ts
-export const myPlugin = createPlugin({
-  id: 'my-plugin',
+```ts title="plugins/tech-radar/src/index.ts"
+export const techRadarPlugin = createPlugin({
+  id: 'tech-radar',
 });
 ```
 
-<!--
+The created plugin does not yet do anything, it will need some plugin options to be useful:
 
-Note that this plugin is useless in itself, and you need to provide extra options to make it useful, in particular extensions.
-
- -->
+- Extend it's functionality, through pages, navigation items or entity cards, using `extensions`
+- You can make it reachable through `routes`
+- Link to other plugins in Backstage using `externalRoutes`
+- Pass `featureFlags` to it
 
 ### Plugin ID
 
-<!--
+You will decide on the `id` when you [create a plugin](../../). The plugin `id` should follow the respective [naming pattern](./08-naming-patterns.md#plugins) of the new frontend system. To give you an example the GraphiQL plugin has the `id` `tech-radar` & as it follows the naming convention we know that the `techRadarPlugin` is the symbol for the default package export containing the plugin.
 
-link to relevant docs
+### Plugin Extensions
 
--->
+So let's make our plugin a bit more useful! Imagine we want to have a TechRadar plugin that displays the recommended technologies at our organisation, [just like in the Backstage demo instance](https://demo.backstage.io/tech-radar). We want the plugin to be displayed on a page & have a navigation item directing to it. This can be achieved by creating a page extension & a navigation item extension. When adding those to the plugin they will be imported with the plugin package & can be discovered.
 
-### Plugin Extensions
+```ts title="plugins/tech-radar/src/index.ts"
+export const techRadarPlugin = createPlugin({
+  id: 'tech-radar',
+  extensions: [techRadarPage, techRadarNavItem],
+});
+```
 
-<!--
+### Plugin Routes
 
- - Example of how this option is used in `createPlugin`
+Now that we have a page extension in our plugin the route of the page is detected & the page is available. Thought to make the route easy accessible through the plugin you are required to additionally add those routes on the plugin level. If we have a `techRadarRootRef`, that is the route reference for the `techRadarPage`, we would add it to `createPlugin` like the following:
 
-link to relevant docs
+```ts title="plugins/tech-radar/src/index.ts"
+export const techRadarPlugin = createPlugin({
+  id: 'tech-radar',
+  extensions: [techRadarPage, techRadarNavItem],
+  routes: {
+    root: techRadarRootRef,
+  },
+});
+```
 
--->
+### Plugin External Routes
```

Comment:
External routes are already explained in the routing architecture docs. I think it's perhaps best to collapse this and the previous section into a single plugin routes docs that covers both, but quickly links over to the routing docs for more details?

Reply:
not sure what you mean by this. should I create a seperate `0x-plugin-routes.md` file in the `architecture` folder?

Reply:
There is one already, it's all documented here: https://backstage.io/docs/frontend-system/architecture/routes/

I still think this section is useful, but that it would be best to treat it just as a very quick introduction to routes and then quickly linking over to the routing architecture docs instead.